builtin/providers/aws/resource_aws_vpn_connection.go

```diff
@@ -0,0 +1,356 @@
+						},
+
+						"source": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+
+						"state": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(fmt.Sprintf("%s-", m["destination_cidr_block"].(string)))
+					buf.WriteString(fmt.Sprintf("%s-", m["source"].(string)))
+					buf.WriteString(fmt.Sprintf("%s-", m["state"].(string)))
+					return hashcode.String(buf.String())
+				},
+			},
+
+			"vgw_telemetry": &schema.Schema{
+				Type:     schema.TypeSet,
+				Computed: true,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"accepted_route_count": &schema.Schema{
+							Type:     schema.TypeInt,
+							Computed: true,
+							Optional: true,
+						},
+
+						"last_status_change": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+
+						"outside_ip_address": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+
+						"status": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+
+						"status_message": &schema.Schema{
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(fmt.Sprintf("%s-", m["outside_ip_address"].(string)))
+					return hashcode.String(buf.String())
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsVpnConnectionCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	connectOpts := &ec2.VPNConnectionOptionsSpecification{
+		StaticRoutesOnly: aws.Boolean(d.Get("static_routes_only").(bool)),
+	}
+
+	createOpts := &ec2.CreateVPNConnectionInput{
+		CustomerGatewayID: aws.String(d.Get("customer_gateway_id").(string)),
+		Options:           connectOpts,
+		Type:              aws.String(d.Get("type").(string)),
+		VPNGatewayID:      aws.String(d.Get("vpn_gateway_id").(string)),
+	}
+
+	// Create the VPN Connection
+	log.Printf("[DEBUG] Creating vpn connection")
+	resp, err := conn.CreateVPNConnection(createOpts)
+	if err != nil {
+		return fmt.Errorf("Error creating vpn connection: %s", err)
+	}
+
+	// Store the ID
+	vpnConnection := resp.VPNConnection
+	d.SetId(*vpnConnection.VPNConnectionID)
+	log.Printf("[INFO] VPN connection ID: %s", *vpnConnection.VPNConnectionID)
+
+	// Wait for the connection to become available. This has an obscenely
+	// high default timeout because AWS VPN connections are notoriously
+	// slow at coming up or going down. There's also no point in checking
+	// more frequently than every ten seconds.
```

Comment:
Thanks for this extra context in the comment. Super helpful for future humans in here.